src/cursor/cur_std.c

```diff
@@ -1290,6 +1290,38 @@ __wt_cursor_bound(WT_CURSOR *cursor, const char *config)
     API_END_RET_STAT(session, ret, cursor_bound);
 }
 
+/*
+ * __wt_cursor_bounds_save --
+ *     Save cursor bounds to restore the state.
+ */
+void
+__wt_cursor_bounds_save(WT_CURSOR *cursor, WT_CURSOR_BOUNDS_STATE *state)
+{
+    WT_ITEM_SET(state->lower_bound, cursor->lower_bound);
+    WT_ITEM_SET(state->upper_bound, cursor->upper_bound);
+    state->flags = cursor->flags;
+}
+
+/*
+ * __wt_cursor_bounds_restore --
+ *     Restore the cursor's bounds state.
+ */
+void
+__wt_cursor_bounds_restore(WT_CURSOR *cursor, WT_CURSOR_BOUNDS_STATE *bounds_state)
+{
+    uint64_t mask;
+
+    mask = WT_CURSTD_BOUND_UPPER | WT_CURSTD_BOUND_UPPER_INCLUSIVE | WT_CURSTD_BOUND_LOWER |
+      WT_CURSTD_BOUND_LOWER_INCLUSIVE;
+
+    bounds_state->flags = cursor->flags & mask;
```

Comment:
This should just be `bound_state->flags &= mask;`

Reply:
Thanks!